src/main/php/predicate/predicates.php

```diff
@@ -259,6 +259,18 @@ function isNotAnArray(): Predicate
   return not(isArray());
 }
 
+/**
+ * returns predicate which tests for array subset
+ *
+ * @api
+ * @param array<mixed> $other
+ * @since 6.2.0
+ */
+function arraySubset(array $other, bool $strict = false): ArraySubset
```

Comment:
For a proper grammar, I'd rather see something like `containsSubset` so that a line forms a nice sentence:
`assertThat($something, containsSubset(['foo', 'bar']));`

Also, I dislike having boolean arguments at this place because a reader has to be familiar with the boolean argument to be able to understand it. I'd go for something like 
`assertThat($something, containsSubset(['foo', 'bar'])->withStrictCheck());` where this is a method on the returned object. Not sure about that method name, maybe there are better ones, but I think it's better for the reader than a boolean argument.